src/sortedDict.jl

```diff
@@ -256,22 +260,55 @@ immutable SDKeyIteration{T <: SDIterableTypesBase}
     base::T
 end
 
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{SortedDict{K,D,Ord}}) = K
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{ExcludeLast{K,D,Ord}}) = K
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{IncludeLast{K,D,Ord}}) = K
+
 immutable SDValIteration{T <: SDIterableTypesBase}
     base::T
 end
 
+eltype{K,D,Ord <: Ordering}(s::SDValIteration{SortedDict{K,D,Ord}}) = D
+eltype{K,D,Ord <: Ordering}(s::SDValIteration{ExcludeLast{K,D,Ord}}) = D
+eltype{K,D,Ord <: Ordering}(s::SDValIteration{IncludeLast{K,D,Ord}}) = D
```

Comment:
Same notes as above; here, you would need both `K` and `D`, but not `Ord`, since without `K`, there's no way to specify the second type parameter.